### PR TITLE
[operator] Bump operator to 0.77.0

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Operator 0.77.0 added support for Go auto-instrumentation.  As a result, this PR adds:
- the ability to set the go image repo and tag.
- the ability to set feature-gates flag.

### CODE REVIEW

File: charts/opentelemetry-operator/values.yaml [hunk 1]
```diff
@@ -49,6 +49,12 @@ manager:
     dotnet:
       repository: ""
       tag: ""
+    # The Go instrumentaiton support in the operator is disabled by default.
```
Comment (Allex1) [inline charts/opentelemetry-operator/values.yaml:52]:
```suggestion
    # The Go instrumentation support in the operator is disabled by default.
```

File: charts/opentelemetry-operator/values.yaml [hunk 2]
```diff
@@ -49,6 +49,12 @@ manager:
...
+    # To enable it, use the +operator.autoinstrumentation.go feature gate.
```
Comment (Allex1) [inline charts/opentelemetry-operator/values.yaml:53]:
```suggestion
    # To enable it, use the operator.autoinstrumentation.go feature gate.
```

File: charts/opentelemetry-operator/values.schema.json [hunk 3]
```diff
@@ -365,6 +396,14 @@
                         }
                     }]
                 },
+                "featureGates": {
+                    "type": "string",
+                    "default": "+operator.autoinstrumentation.go",
```
Comment (Allex1) [inline charts/opentelemetry-operator/values.schema.json:401]:
```suggestion
                    "default": "operator.autoinstrumentation.go",
```

**Review by Allex1 — APPROVED** [review]

**Review by JaredTan95 — APPROVED** [review]
great works

File: charts/opentelemetry-operator/values.yaml [hunk 4]
```diff
@@ -49,6 +49,14 @@ manager:
...
       repository: ""
       tag: ""
+    # The Go instrumentaiton support in the operator is disabled by default.
+    # To enable it, use the operator.autoinstrumentation.go feature gate.
+    go:
+      # Must set the default value manually for now.
+      # See https://github.com/open-telemetry/opentelemetry-operator/issues/1756 for details.
+      repository: ghcr.io/open-telemetry/opentelemetry-go-instrumentation/autoinstrumentation-go
+      tag: v0.2.1-alpha
+  featureGates: ""
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply charts/opentelemetry-operator/values.yaml:63]: Added.

**Pull request merged**